cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -411,7 +424,6 @@ DatasetAdminOpResponse executeAdmin(DatasetId instance, String method) throws Ex
     Principal principal = authenticationContext.getPrincipal();
     switch (method) {
       case "exists":
-        ensureAccess(instance);
```

Comment:
Since we do not want to expose the dataset instance exists or not if the user does not have any privilege on the dataset. The other two methods will throw a `NotFoundException` if the dataset does not exist. We do not want to expose this info to user if he does not have any privilege